### PR TITLE
Use a shared async gateway between all async apis

### LINKED ISSUE #2279: Create a shared async gateway deployment which runs on the operator node group

### PR DESCRIPTION
**Pull request by vishalbollu**
closes #2279 

---

checklist:

- [x] run `make test` and `make lint`
- [x] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

**Review by miguelvr — APPROVED** [review]
LGTM

File: manager/manifests/async-gateway.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            httpGet:
+              path: /healthz
+              port: 8888
+              scheme: HTTP
+          resources:
+            requests:
+              cpu: 100m
+              memory: 200Mi
+            limits:
+              cpu: 100m
```
Comment (deliahu) [inline manager/manifests/async-gateway.yaml.j2:73]: I think we could make these a little larger, let's discuss live

File: pkg/async-gateway/service.go [hunk 2]
```diff
@@ -78,7 +77,8 @@ func (s *service) CreateWorkload(id string, payload io.Reader, headers http.Head
 	}
 
 	log.Debug("sending message to queue")
-	if err := s.queue.SendMessage(id, id); err != nil {
+	queue := NewSQS(queueURL, &s.session)
```
Comment (deliahu) [inline pkg/async-gateway/service.go:80]: Is this efficient / ok to do on each request?

**Review by deliahu — APPROVED** [review]

File: pkg/async-gateway/service.go (hunk 2 above)
Comment (vishalbollu) [reply pkg/async-gateway/service.go:80]: Yes, this isn't making any IO calls

**Pull request merged**